src/inference/src/core.cpp

```diff
@@ -11,31 +11,46 @@
 #include "openvino/core/so_extension.hpp"
 #include "openvino/runtime/device_id_parser.hpp"
 #include "openvino/runtime/iremote_context.hpp"
+#include "openvino/util/common_util.hpp"
 #include "openvino/util/file_util.hpp"
 
 namespace ov {
 
 std::string find_plugins_xml(const std::string& xml_file) {
-    if (xml_file.empty()) {
-        const auto ov_library_path = ov::util::get_ov_lib_path();
+    std::string xml_file_name = xml_file;
+    if (xml_file_name.empty()) {
+        // Default plugin xml file name.
+        xml_file_name = "plugins.xml";
+    } else {
+        // Check plugin xml extension name.
+        if (!ov::util::ends_with(xml_file_name, "xml") && !ov::util::ends_with(xml_file_name, "XML")) {
+            OPENVINO_THROW("Unsupport plugin xml file with non-xml extension name(.xml or .XML): ",
+                           xml_file_name.c_str());
+        }
```

Comment:
The check allows search with any type of path inside OV folder, relative paths are harmful